client/my-sites/people/role-select/index.jsx

```diff
@@ -2,138 +2,87 @@
-import RolesActions from 'lib/site-roles/actions';
 import FormFieldset from 'components/forms/form-fieldset';
 import FormLabel from 'components/forms/form-label';
 import FormSelect from 'components/forms/form-select';
 import FormSettingExplanation from 'components/forms/form-setting-explanation';
-import sitesList from 'lib/sites-list';
-
-/**
- * Module variables
- */
-const debug = debugFactory( 'calypso:my-sites:people:role-select' );
-const sites = sitesList();
-
-export default React.createClass( {
-	displayName: 'RoleSelect',
-
-	getInitialState() {
-		return ( {
-			roles: this.props.siteId ? RolesStore.getRoles( this.props.siteId ) : {}
-		} );
-	},
-
-	componentWillMount() {
-		if ( this.props.includeFollower ) {
-			this.refreshRoles();
-		}
-	},
-
-	componentDidMount() {
-		RolesStore.on( 'change', this.refreshRoles );
-		this.fetchRoles();
-	},
-
-	componentWillUnmount() {
-		RolesStore.removeListener( 'change', this.refreshRoles );
-	},
-
-	componentWillReceiveProps( nextProps ) {
-		const siteId = nextProps.siteId || this.props.siteId;
-		this.fetchRoles( siteId );
-		this.refreshRoles( nextProps );
-	},
-
-	getWPCOMFollowerRole( siteId ) {
-		siteId = siteId ? siteId : this.props.siteId;
-
-		const site = sites.getSite( siteId ),
-			displayName = site.is_private
-			? this.translate( 'Viewer', { context: 'Role that is displayed in a select' } )
-			: this.translate( 'Follower', { context: 'Role that is displayed in a select' } );
-
-		return {
-			follower: {
-				display_name: displayName,
-				name: 'follower'
-			}
-		};
-	},
-
-	refreshRoles( nextProps ) {
-		const siteId = nextProps && nextProps.siteId ? nextProps.siteId : this.props.siteId;
-
-		if ( siteId ) {
-			let siteRoles = RolesStore.getRoles( siteId );
-
-			if ( this.props.includeFollower ) {
-				siteRoles = Object.assign( {}, this.getWPCOMFollowerRole(), siteRoles );
-			}
-
-			this.setState( {
-				roles: siteRoles
-			} );
-		}
-	},
-
-	fetchRoles( siteId = this.props.siteId ) {
-		if ( ! siteId ) {
-			debug( 'siteId not set' );
-			return;
-		}
-
-		if ( Object.keys( RolesStore.getRoles( siteId ) ).length ) {
-			debug( 'initial fetch not necessary' );
-			return;
-		}
-
-		debug( 'Fetching roles for ' + siteId );
+import QuerySites from 'components/data/query-sites';
+import QuerySiteRoles from 'components/data/query-site-roles';
+import { getSite } from 'state/sites/selectors';
+import { getSiteRoles } from 'state/site-roles/selectors';
+
+const getWPCOMFollowerRole = ( { site, translate } ) => {
+	const displayName = site.is_private
+		? translate( 'Viewer', { context: 'Role that is displayed in a select' } )
+		: translate( 'Follower', { context: 'Role that is displayed in a select' } );
+
+	return {
+		display_name: displayName,
+		name: 'follower'
+	};
+};
+
+const RoleSelect = ( props ) => {
+	const { siteRoles, site, includeFollower, siteId, id, explanation, translate } = props;
+	const roles = siteRoles && siteRoles.slice( 0 );
```

Comment:
Instead of `Array#slice` to copy the array because we use `Array#push` below, we could instead use `Array#concat`:

```js
let { siteRoles } = props;
if ( siteRoles && includeFollower ) {
	siteRoles = siteRoles.concat( getWPCOMFollowerRole( props ) );
}
```